### PR TITLE
Add `expires_in` and `issued_at` to the auth spec.

### PR DESCRIPTION
**Pull request by mattmoor**
This proposes an extension to the specification for the Bearer token
exchange to include information pertaining to when an issued Bearer
token will expire.

Signed-off-by: Matt Moore <mattmoor@google.com>

### CODE REVIEW

**Comment by mattmoor** [comment]
@jlhawn @stevvooe @aaronlehmann FYI

File: docs/spec/auth/token.md [hunk 1]
```diff
@@ -159,15 +189,16 @@ It is this intersected set of access which is placed in the returned token.
 
 The server then constructs an implementation-specific token with this
 intersected set of access, and returns it to the Docker client to use to
-authenticate to the audience service:
+authenticate to the audience service (the indicated window of time):
```
Comment (jlhawn) [inline docs/spec/auth/token.md]: `*within* the indicated window of time`?

**Comment by jlhawn** [comment]
I think there's a minor typo, but otherwise LGTM

File: docs/spec/auth/token.md (hunk 1 above)
Comment (mattmoor) [inline docs/spec/auth/token.md]: ofc, thanks!

**Comment by mattmoor** [comment]
@dmcgowan FYI

Let me know if folks are alright with this, and I'm happy to put together some code for it (likely borrowing heavily from my original PR).

**Comment by dmcgowan** [comment]
LGTM

**Comment by mattmoor** [comment]
Wondering what the next steps are here, should I just augment this with the code change?  Are there other folks who should weigh in and LGTM?

@stevvooe Thoughts?

**Comment by RichardScothern** [comment]
LGTM

**Comment by mattmoor** [comment]
Ping?

**Comment by RichardScothern** [comment]
@mattmoor if you PR your code change, I'll merge these together to keep the code / documentation in sync.

**Comment by mattmoor** [comment]
I have updated this with the code from my prior PR, which has been updated to include the `issued_at` logic (and testing for it).

**Comment by mattmoor** [comment]
@RichardScothern Just wanted to bump this, I added the code to this PR last week.

**Comment by RichardScothern** [comment]
Code LGTM.  Ping @dmcgowan @stevvooe @aaronlehmann

**Comment by stevvooe** [comment]
This is looking good. Let's just add the "access_token" alias for "token".

**Comment by mattmoor** [comment]
@stevvooe I have updated the `token.md` with the changes to the specification in a separate commit.  I will update this commit with the code shortly (and can collapse before merging the PR).

**Comment by mattmoor** [comment]
I updated the code as well

**Comment by mattmoor** [comment]
Just want to bump this.  Was there anything else?  Should I squash the commits?

**Comment by mattmoor** [comment]
@stevvooe Wanted to bump this before today's DWG meeting in case there was anything further to discuss?

File: docs/spec/auth/token.md [hunk 2]
```diff
@@ -113,6 +113,45 @@ challenge, the client will need to make a `GET` request to the URL
...
+        <code>expires_in</code>
+    </dt>
+    <dd>
+        (Optional) The duration in seconds since the token was issued that it 
+        will remain valid.  When omitted, this defaults to 60 seconds.  For
+        compatibility with older clients, a token should never be returned with
+        less than 60 seconds to live.
+    </dd>
+    <dt>
+        <code>issued_at</code>
```
Comment (stevvooe) [inline docs/spec/auth/token.md]: What is the time format used here?

**Comment by stevvooe** [comment]
@mattmoor These changes are everything we talked about. We just need to add the correct time format for the `issued_at` field.

**Comment by mattmoor** [comment]
Done and collapsed the commits.

**Comment by stevvooe** [comment]
LGTM

@jlhawn PTAL

**Comment by mattmoor** [comment]
@jlhawn Monday bump.

**Comment by jlhawn** [comment]
LGTM

**Comment by RichardScothern** [comment]
please rebase @mattmoor  to pick up the golint fix.

**Comment by mattmoor** [comment]
Done, thanks.

**Pull request merged**